modules/eventlog/src/main/java/org/apache/ignite/internal/eventlog/impl/ConfigurationBasedChannelRegistry.java

```diff
@@ -35,73 +34,57 @@
 import org.apache.ignite.internal.eventlog.config.schema.ChannelView;
 import org.apache.ignite.internal.eventlog.config.schema.EventLogConfiguration;
 
-class ConfigurationBasedChannelRegistry implements ChannelRegistry {
-    private final ReadWriteLock guard;
+public class ConfigurationBasedChannelRegistry implements ChannelRegistry {
+    private final AtomicReference<Map<String, EventChannel>> cache;
 
-    private final Map<String, EventChannel> cache;
-
-    private final Map<String, Set<EventChannel>> typeCache;
+    private final AtomicReference<Map<String, Set<EventChannel>>> typeCache;
 
     private final SinkRegistry sinkRegistry;
 
-    ConfigurationBasedChannelRegistry(EventLogConfiguration cfg, SinkRegistry sinkRegistry) {
-        this.guard = new ReentrantReadWriteLock();
-        this.cache = new HashMap<>();
-        this.typeCache = new HashMap<>();
+    public ConfigurationBasedChannelRegistry(EventLogConfiguration cfg, SinkRegistry sinkRegistry) {
+        this.cache = new AtomicReference<>(new HashMap<>());
+        this.typeCache = new AtomicReference<>(new HashMap<>());
         this.sinkRegistry = sinkRegistry;
 
         cfg.channels().listen(new CacheUpdater());
     }
 
     @Override
     public EventChannel getByName(String name) {
-        guard.readLock().lock();
-        try {
-            return cache.get(name);
-        } finally {
-            guard.readLock().unlock();
-        }
+        return cache.get().get(name);
     }
 
     @Override
     public Set<EventChannel> findAllChannelsByEventType(String igniteEventType) {
-        guard.readLock().lock();
-        try {
-            Set<EventChannel> result = typeCache.get(igniteEventType);
-            return result == null ? Set.of() : new HashSet<>(result);
-        } finally {
-            guard.readLock().unlock();
-        }
+        Set<EventChannel> result = typeCache.get().get(igniteEventType);
+        return result == null ? Set.of() : new HashSet<>(result);
```

Comment:
No need to copy `result`.

Reply:
Why? I return the collection outside and I do not want anybody to modify it.